src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,50 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  removeTodo:(todoId: number) => void;
+  updateState:(todoId: number, property: Partial<Todo>) => void;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  removeTodo,
+  updateState,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => {
+        const { id, title, completed } = todo;
+
+        return (
+          <div
+            data-cy="Todo"
+            className={classNames('todo', { completed })}
+            key={id}
+          >
+            <label className="todo__status-label">
+              <input
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+                onChange={() => updateState(id, { completed: !completed })}
```

Comment:
create a separate function for this event. and update it on the server and locally as well